### ISSUE TITLE
Auto scroll to nearest 'viewable' patch

### ISSUE THREAD

**Issue by tkeyes25** [utk-se/CodeRibbon#26]
When user scrolls around, be able to load up the nearest set of patches automatically

**Comment by robobenklein** [comment]
https://stackoverflow.com/questions/123999/how-to-tell-if-a-dom-element-is-visible-in-the-current-viewport

There is a non-jQuery listener version that we should explore there

**Comment by robobenklein** [comment]
scroll patch into center of view works: https://github.com/utk-se/CodeRibbon/pull/25/files#diff-0c38785aa45d91cacd92d3ff95aa20afR55

does not work for even number of patches, and if using start/end/nearest instead of center it often ends up with partially visible patches

**Comment by tkeyes25** [comment]
Where did you leave off on this? I know ```scrollIntoView()``` was something we had explored beforehand

**Comment by tkeyes25** [comment]
I see the commit, left a comment https://github.com/utk-se/CodeRibbon/pull/25/files#r259457220

**Comment by tkeyes25** [comment]
HELP: How do you get more than 6 patches? Hahah. I wanted to tackle #18 but cannot even get to the horizontal scrolling on multiple patches

**Comment by robobenklein** [comment]
Yeah that's exactly how you add more... increase then decrease the number in settings lol

**Comment by tkeyes25** [comment]
Hahah alright, we can look into that as another issue

**Comment by robobenklein** [comment]
#32 will make this easier

**Comment by robobenklein** [comment]
Research required: https://github.com/lucafalasco/scroll-snap
